driver/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -41,6 +41,8 @@
     private int bufferOffset;
     private long currentPosition;
     private byte[] buffer = null;
+    private int markReadLimit;
```

Comment:
Its really murky and unclear. 

To work with the general contract, I'm honoring the set limit - else why set one? I have just added a `mark()` helper that I expect will actually be used by anyone wanting to `mark` and `reset`.